### PR TITLE
Milestone 2: implementing componets, reducers, actions and Store

### PR DESCRIPTION
**Pull request by nicolasmarino99**
This milestone was focused on understanding the main React-redux library functionalities. The following objectives were achieved on each technology respectively as follows:

#### components 🔌
- App component which wrapped BooksList and BooksForm components.
- Implementation of the BooksList component which connects to the Redux store and fetches the books from the state.
- Implementation of the Book component.
- Implementation of BooksForm component.

#### Reducers 🔌
- Implementation of a Book reducer wich Removes and adds books from/to the applications state.
#### Actions 🔌
- Implementation of  (remove and create) Book actions for passing current state data to reducers applications state.
#### Store 🔌
- Implementation of a redux store for receiving the created Reducers and mock state for checking app current functionalities.

### CODE REVIEW

File: src/components/BooksForm.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import categories from '../variables/Categories';
+
+const BookForm = () => (
+  <form>
```
Comment (Torres-ssf) [inline src/components/BooksForm.js]: - [x] As it said in the instructions, we need to Separate components in Presentational Components and Container Componentes. This approach can make your app easier to understand and allow you to more easily reuse components. In your components directory (presentational components) you have components that are not connected to the Redux store, and on your containers directory, you have components that are connected to the Redux store. This is the common approach followed when using Redux with React. Check out [here](https://redux.js.org/basics/usage-with-react#presentational-and-container-components). Right now BookForm does not connect with Redux yet, but we will do that in a future milestone. We need to move BookForm from `components` directory to the `containers` directory.
Screenshot from 2020-08-18 19-36-52

File: src/components/BooksList.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import Book from './Book';
+
+// eslint-disable-next-line react/prop-types
+const BookList = ({ books }) => (
+  <table>
+    <thead>
+      <tr>
+        <th>Book ID</th>
+        <th>Title</th>
+        <th>Category</th>
```
Comment (Torres-ssf) [inline src/components/BooksList.js]: - [x] We need to move BookForm from `components` directory to the `containers` directory.

File: src/index.js [hunk 3]
```diff
@@ -1,13 +1,28 @@
...
 import * as serviceWorker from './serviceWorker';
+import booksReducer from './reducers/index';
+
+const state = {
+  books: [
+    { id: Math.floor(1000 * Math.random(5, 5)), title: 'Lolita', category: 'Romance' },
+    { id: Math.floor(1000 * Math.random(5, 5)), title: 'Lolita', category: 'Romance' },
+    { id: Math.floor(1000 * Math.random(5, 5)), title: 'Lolita', category: 'Romance' },
+    { id: Math.floor(1000 * Math.random(5, 5)), title: 'Lolita', category: 'Romance' },
+    { id: Math.floor(1000 * Math.random(5, 5)), title: 'Lolita', category: 'Romance' },
```
Comment (Torres-ssf) [inline src/index.js]: **OPTIONAL**

- [x] You can use uuidv4 to automatically generate ids for you. Check the package [here](https://www.npmjs.com/package/uuidv4).
```javascript
import uuid from 'uuidv4'

 { id: uuid(), title: 'Lolita', category: 'Romance' }, //generates an unique uuid.
```

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
### Changes required :muscle: 
Hi @nicolasmarino99  :hand: 
Nice job so far on this milestone :clap: 
But before moving, we need to work on some changes.

- [x] Check my comments bellow

Feel free to message me on slack if you have any questions :smile: 
Happy Coding :computer: 
**Sergio Torres**

File: src/components/BooksForm.js (hunk 1 above)
Comment (nicolasmarino99) [reply src/components/BooksForm.js]: Thanks Sergio for taking the time to explain these concepts. :)

File: src/index.js (hunk 3 above)
Comment (nicolasmarino99) [reply src/index.js]: Thanks for the suggestion. 👍

**Review by bellom — APPROVED** [review]
## Hi @nicolasmarino99 :clap: 

**Great work making all the necessary corrections from the previous review.**

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**